core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/HeaderMarshaller.java

```diff
@@ -79,4 +72,36 @@ public void marshall(T val, JsonMarshallerContext context, String paramName, Sdk
             context.request().appendHeader(paramName, converter.convert(val, sdkField));
         }
     }
+
+    private static class SimpleListHeaderMarshaller implements JsonMarshaller<List<?>> {
+        private Map<Class<?>, Predicate> skipPredicateMap = new HashMap<>();
+
+        {
+            skipPredicateMap.put(String.class,
+                                 k -> k instanceof String && StringUtils.isBlank((String) k));
+        }
```

Comment:
Not sure I see the need for this map, at the moment since they're only predicate in the map.

`shouldSkipElement` can be simplified to just

```java
listValue instanceof String && StringUtils.isBlank((String) listValue)
```

Reply:
Agree ,sine we have do not have complex shapes and we have only predefined set of Classes for Header marhsalling like https://github.com/aws/aws-sdk-java-v2/blob/55f314fbe4af9cfac0c67cbff505476a5c344a5f/core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/marshall/JsonProtocolMarshaller.java#L114-L123 , directly checking for String without keeping a map based on types looks fine. Will update this